test/parallel/test-worker-max-count-disabled.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { spawn } = require('child_process');
+const { Worker } = require('worker_threads');
+const { cpus } = require('os');
+
+if (process.argv[2] === 'child') {
+  const cpu_count = cpus().length;
+
+  // This may need to be updated in the future if there are
+  // any other warnings expected. For now, no warnings are
+  // emitted with this code.
+  process.on('warning', common.mustNotCall());
+
+  const expr = `
+    setInterval(() => {}, 10);
+    require("worker_threads").parentPort.postMessage({});
+  `;
+
+  function makeWorker(workers) {
+    return new Promise((res) => {
+      const worker = new Worker(expr, { eval: true });
+      worker.on('message', res);
+      workers.push(worker);
+    });
+  }
+
+  async function doTest() {
+    const workers = [];
+    const list = [];
+    for (let n = 0; n < cpu_count; n++)
+      list.push(makeWorker(workers));
+    await Promise.all(list);
+    workers.forEach((i) => i.terminate());
+  }
+
+  doTest().then(common.mustCall());
+
+} else {
+  const child = spawn(
+    process.execPath,
+    ['--max-worker-threads=0', __filename, 'child'],
+    { stdio: 'inherit' });
+  child.on('close', common.mustCall((code) => {
+    assert.strictEqual(code, 0);
+  }));
+}
```

Comment:
I don’t think we need to spawn child processes here, if all we’re doing is asserting that the process exits cleanly?

Reply:
Yeah, I did it like this to ensure that the test would work even if someone happened to have `--max-worker-threads=n` set to some other value in a `NODE_OPTIONS` environment on their system.

Reply:
Switching this to use the flags header at the top and removing the child process